### DISCUSSION TITLE
"Fixed amounts" will be removed ? Answer: NO!

### DISCUSSION THREAD

**Discussion by Medius7** [kimai/kimai#3127] [Questions & Answers] [answered]
Hello,

I want to avoid going offtopic in the issue tracker, so I ask here:

@kevinpapst  wrote:
> And be warned: fixed amounts are already planned to be removed from the next major version

I'm often using fixed amounts in kimai to track completed tasks / jobs, for which I have agreed a fixed amount with the client.

I didnt found anything about the removal of fixed amounts in kimai. Can you point me anywhere, or explain, why this will be removed? Is there any upcoming feature, which replaces this features?

**Comment by kevinpapst** [comment, marked as answer]
❗❗❗  **FIXED AMOUNTS WILL NOT BE REPLACED (see below)** ❗❗❗

Fixed amount can be entirely replaced by the Expense plugin. Yes, that costs a one time fee. But you are managing your income with Kimai, so I hope this is too much to ask for.

Fixed amounts are mostly (in over 90% of all cases) NOT used in the way they were intended.
Instead you are using a timesheet record to trigger a certain reaction in the invoice system.

I can tell you that the database column for fixed amount is not filled in over 99,99% of all rows. In all systems that I saw - and I saw many... Every piece of code that exists in the system and that is not used or that does not match the Pareto principle should be questioned (especially when it comes to the timesheet table). 
Even if fixed amounts are sometimes used in the way they were intended, it can now be done with expenses as well. 

So there is a full and better replacement for that functionality.

> I didnt found anything about the removal of fixed amounts in kimai.

 That is not correct, you are here because you found something about it^^

I am not sharing every change I plan, correct. As steward of this ship I am talking to multiple customers (Kimai users) each and every day, many are freelancer or small companies, some are bigger and some are very big (multiple hundred users in Kimai). 
I am making decisions mainly based on that input and based on votings here on Kimai. Single voices/votings are usually not to be trusted in decision making. A rule which I apply to myself as well! If I would only add features I need, Kimai would look entirely different.

**Reply by kevinpapst** [reply]
And btw, you can checkout the future version here: https://github.com/kevinpapst/kimai2/pull/2902

**Reply by Medius7** [reply]
Thanks for the answer. Of course, I'm glad I found Kimai, and I'm thankful for great work.

I'm using the Kimai.cloud. Will the "Expense plugin" be avaiable there? Or is it already active? :-)

**Reply by kevinpapst** [reply]
Ah, okay. I didn't anticipate that Cloud users come here to discuss. Usually the ones being active at Github are the ones self-hosting. In that case: the expense plugin is available, but only in a paid plan. 

And to make the information complete: you can still create an entry with one hour and set the hourly rate to the fixed amount, as workaround for the missing fixed rate in the next major release.

**Comment by Sarmingsteiner** [comment]
I was also very anxious when I read this. Ok, the expense plugin should replace it, but it will be more cumbersome:

To also record the time spent on the tasks billed with (formerly) fixed amounts, IMHO we need to:
1) Create a timesheet, mark it as not billable and set the rate to zero (otherwise, it would be included in the daily sum-up).
2) Create a new expense

It is not so much about the money, but it is more complicated than before and we have to educate/watch our employees.

**Comment by kevinpapst** [comment]
After all the feedback here I have been talking to many more customers / users of Kimai about "fixed rates".

And even though I'd like to remove this functionality (due to technical reasons), I will not: 

❗Fixed amounts functionality will be kept❗

I am listening to the community and you raised your voice: thanks for that 👍

**Reply by Medius7** [reply]
Nice to hear this. Thank you for changing your mind :-)
